project-reports/2023/2023-Q4-Hyperledger-Fabric.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: default
+title: 2023 Q4 Hyperledger Fabric
+parent: 2023
+grand_parent: Project Updates
+has_children: false
+has_toc: false
+---
+
+# 2023 Q4 Hyperledger Fabric
+
+# Project Health
+
+Hyperledger Fabric is fairly mature and stable with a v2.5 Long-term support (LTS) release.
+There is less churn and fewer commits than in years past, with continued focus on quality and support.
+New features get proposed, approved, and implemented based on a community RFC process.
+Mailing list activity is down a bit compared to prior years.
+PRs and mailing list questions are generally turned around quickly.
+There is significant Discord activity.
+
+# Questions/Issues for the TOC
+
+None.
+
+# Releases
+
+LTS releases:
+- v2.5 is the current LTS release with patch releases approximately quarterly.
+- v2.2 is the prior LTS releases, critical fixes will be backported through 2023.
+
+The Fabric [v2.5 announcement](https://lists.hyperledger.org/g/fabric/message/11754) encouraged users to upgrade to v2.5 so that they can keep getting fixes and updates that will eventually be targeted for v2.5.x releases only.
+The announcement also indicated that the legacy Fabric SDKs (known as Fabric SDK for Node v2.2, Fabric SDK for Java v2.2, Fabric SDK for Go v1.0) are deprecated.
+Users of the legacy SDKs are encouraged to evaluate the Fabric Gateway client API as a replacement.
+The Fabric Gateway client API is the recommended SDK for client applications when using Fabric v2.4 or v2.5, and is available in Go, Node, and Java programming languages.
+
+[Releases](https://github.com/hyperledger/fabric/releases) past quarter:
+
+- Fabric v2.2.13 - August 2, 2023
+- Fabric v2.5.3 - June 14, 2023
+- Fabric v2.5.4 - August 2, 2023
+- Fabric v3.0.0-preview - September 1, 2023
+
+# Overall Activity in the Past Quarter
+
+The project delivered fixes to the Fabric v2.2 (prior LTS) and v2.5 (current LTS) releases. See [release notes](https://github.com/hyperledger/fabric/releases) for details.
+
+A v3.0.0-preview release was created to gather feedback on the new SmartBFT consensus algorithm that is being developed for the Fabric ordering service (main branch).
+
+An initial v0.0.0 of Go [fabric-admin-sdk](https://github.com/hyperledger/fabric-admin-sdk) has been released to help with programatic channel and chaincode management.
+The project is evaluating the use of this admin SDK in a new and improved [fabric-cli](https://github.com/hyperledger/fabric-cli) that may eventually replace the original peer CLI commands.
```

Comment:
These two items were in the last update. Has there been any updates since the last update?

Reply:
The most significant activity has been around the SmartBFT feature for the v3.0.0-preview release. In the prior quarterly update we mentioned that we were working on the samples and documentation for an eventual v3.0.0-preview release. In the past quarter that work was completed and the release was successfully published in order to gather feedback on it.

The fabric-admin-sdk item was indeed a copy/paste from the prior quarterly update. The v0.0.0 release actually took place in this report quarter (July), but it was mentioned in the prior release since that report came out right after the release in July. This is the classic question of whether to report on what happened in the calendar quarter versus what happened in the last 90 days. Since I thought there would be community interest in this item I included it in both reports.

In general, I am guilty of copy/paste between reports if I think the information is still timely and important, or if work is ongoing across quarters. For example the general message about LTS releases, users upgrading, and the fact that v3.0.0 will be brewing for an extended time is important enough to repeat in my opinion.

Reply:
Thanks for the clarification @denyeart. No problems with the repeats, just wanted to be sure I was getting the full scope of activity this past quarter.